run-regression-tests.sh

```diff
@@ -0,0 +1,227 @@
+                                targetBranch="$1"
+                                ;;
+        -bs | --bootstrap )     shift
+                                bootstrap="$1"
+                                ;;
+        -h | --help )           usage
+                                exit
+                                ;;
+        * )                     error "Unexpected argument $1"
+                                usage
+                                exit 1
+    esac
+    shift
+done
+
+if [[ "${bootstrap}" == "" ]]; then
+    error "Need to specify --bootstrap <URI>."
+    usage
+    exit 1
+fi
+
+#-----------------------------------------------------------------------------------------
+# Main logic
+#-----------------------------------------------------------------------------------------
+verifyTool "argocd"
+verifyTool "tkn"
+verifyTool "go"
+
+if [[ ! -e "../cli" ]]; then
+    error "../cli is not present. Clone the CLI repository from https://github.com/galasa-dev/cli"
+    exit 1
+fi
+
+# Create a new Argo CD app if a branch other than main is specified, otherwise just run regression tests from main
+appName="${targetBranch}-maven-repos"
+if [[ "${targetBranch}" != "main" ]]; then
+  h1 "Creating Argo CD app ${appName}"
+  repositories=(
+      'wrapping'
+      'maven'
+      'gradle'
+      'framework'
+      'extensions'
+      'managers'
+      'obr'
+  )
+
+  index=-1
+  for i in "${!repositories[@]}"; do
+    if [[ "${targetRepository}" == "${repositories[$i]}" ]]; then
+      index=$i
+      break
+    fi
+  done
+
+  if [[ ${index} -eq -1 ]]; then
+    error "Repository '${targetRepository}' not found."
+    exit 1
+  fi
+
+  h2 "Logging into Argo CD"
+  argocd login argocd.galasa.dev --sso --grpc-web
+
+  h2 "Creating app"
+  argocd app create "${appName}" \
+                    --project default \
+                    --sync-policy auto \
+                    --sync-option Prune=true \
+                    --repo https://github.com/galasa-dev/automation \
+                    --revision HEAD \
+                    --path infrastructure/ibmcloud-galasadev-cluster/galasa-development/branch-maven-repository \
+                    --dest-server https://kubernetes.default.svc \
+                    --dest-namespace galasa-development \
+                    --grpc-web
+  rc=$? ; if [[ "${rc}" != "0" ]]; then error "Failed to create ${appName}." ; exit 1 ; fi
+
+  h2 "Configuring helm parameters"
+  for repo in "${repositories[@]:${index}}"; do
+      argocd app set "${appName}" \
+        --helm-set "${repo}.branch=${targetBranch}" \
+        --helm-set "${repo}.imageTag=${targetBranch}" \
+        --helm-set "${repo}.deploy=true" \
+        --grpc-web
+  done
+  success "${appName} app created."
+
+  pipeline="branch-${targetRepository}"
+  h1 "Starting ${pipeline} build pipeline"
+  tkn pipeline start "${pipeline}" \
+      --param fromBranch="main" \
+      --param toBranch="${targetBranch}" \
```

Comment:
Hi Eamonn, I have just realised that Wrapping does not have the same fromBranch and toBranch parameters like the other pipelines, it just has branch, so if you start the script using --repo wrapping this will not work. I don't think your script should be changed, I think we need to change the Wrapping pipeline to accept fromBranch and toBranch.